zcash_primitives/src/transaction/mod.rs

```diff
@@ -60,7 +66,7 @@ pub use zcash_protocol::TxId;
 /// transaction fields. Note that this is not dependent on epoch, only on transaction encoding.
 /// For example, if a particular epoch defines a new transaction version but also allows the
 /// previous version, then only the new version would be added to this enum.
```

Comment:
```suggestion
/// previous version, then only the new version would be added to this enum.
///
/// The derivation of `PartialOrd` depends on the versions being in order with
/// `ZFuture` just after the tx version that it is based on.
```

Reply:
This should not have ordering; instead, the correct approach is to define `TxVersion::has_zip233`, and have the implementation of that method internally perform the necessary determination.